src/chain.h

```diff
@@ -325,7 +322,11 @@ class CDiskBlockIndex : public CBlockIndex
             READWRITE(nBits);
             READWRITE(nNonce);
             if(this->nVersion > 3) {
-                READWRITE(mapZerocoinSupply);
+                // zc supply removed in 4.0.99.2
+                if (nSerVersion < DBI_SER_VERSION_NO_ZC) {
+                    std::map<libzerocoin::CoinDenomination, int64_t> mapZerocoinSupply;
+                    READWRITE(mapZerocoinSupply);
```

Comment:
is the write operation ever used here? could be just `READDATA`?

Reply:
wouldn't use basic operations outside serialize.h. We can simply wrap it inside `if (ser_action.ForRead())` as it is done elsewhere.